packages/render-fidelity-tools/src/components/renderers/gltfpt-viewer.ts

```diff
@@ -77,41 +76,35 @@ export class PathtracingViewer extends LitElement {
   private async[$updateScenario](scenario: ScenarioConfig) {
     this[$canvas] = this.shadowRoot!.querySelector('canvas');
 
-    const dpr = resolveDpr();
-
-    const settings = {
-      pathtracing: true,
-      debugMode: 'None',
-      maxBounceDepth: 1,
-      useIBL: true,
-      autoScaleOnImport: false,
-      pixelRatio: 1.0 / dpr,  // TODO: clarify
-      autoRotate: false,
-      disableDirectShadows: true
-    };
-
-    this[$renderer] =
-        new PathtracingRenderer({'canvas': this[$canvas]!}, settings);
+    const enableControls = false;
+    this[$renderer] = new PathtracingRenderer(this[$canvas]!, enableControls);
 
     const renderer = this[$renderer];
+    renderer.autoScaleOnImport(false);
+    renderer.setPixelRatio(0.5);
+    renderer.setMaxBounceDepth(8);
 
     await new Promise((resolve) => {
       console.log('Loading resources for', scenario.model);
       renderer.loadScene(scenario.model, scenario.lighting, () => {
         this[$updateSize]();
         // console.log(this[$canvas]!.width, this[$canvas]!.height);
-        renderer.resize(this[$canvas]!.width, this[$canvas]!.height)
+        renderer.resize(this[$canvas]!.width, this[$canvas]!.height);
         resolve();
       });
     });
 
-    const numSamples = 8;
+    let numSamples = scenario.pt?.numSamples;
 
-    console.log('Rendering...');
-    renderer.render(
+    if (numSamples == null) {
+      numSamples = 1024;
+    }
+
+    console.log('Rendering ' + numSamples + ' samples');
+    this[$renderer].render(
         numSamples,
-        () => {
-          console.log('frame finished');
+        (frame: number) => {
+          console.log('frame finished ' + frame);
         },
         () => {
           // Wait two rAFs to ensure we rendered at least once //TODO: clarify
```

Comment:
We recently found that two rafs was unnecessary in our other renderers, so it likely is here too. I wonder if this is actually causing you to render the whole scene twice, which would be unfortunate.

Reply:
Ok, that was really just copy&paste, didn't think much about it. I'll check.